app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,63 @@
+class AlbumsController < ApplicationController
+
+  def get_album
+    @album = Album.find(params[:id])
+  end
+
+  def index
+    @albums = Album.all
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def edit
+    get_album
+  end
+
+  def show
+    get_album
+  end
+
+  def update
+    if params[:class] == "upvote"
+      get_album
```

Comment:
Minor refactor, you could bump ```get_album``` up above the beginning of your if statement since it needs to be called regardless of the :class